.github/workflows/release-e2e.yaml

```diff
@@ -0,0 +1,57 @@
+---
+name: release-e2e
+
+on:
+  release:
+    types: [created]
+  workflow_dispatch:
+
+jobs:
+  release-e2e-kind:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '~1.18'
+
+    - name: install_kind_binary
+      env:
+        KIND_VERSION: v0.12.0
+        KIND_OS: kind-linux-amd64
+      run: |
+        curl -sLo kind "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/${KIND_OS}"
+        chmod +x kind
+        sudo mv kind /bin/
+
+    - name: start_kind_cluster
+      run: |
+        /bin/kind create cluster
+        /bin/kind export kubeconfig
+
+    - name: install_kubectl
+      run: |
+        curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
```

Comment:
Nit: should we think about pinning this based on the contents of go.mod (ala https://github.com/operator-framework/rukpak/blob/2d3120c7f7c97ab52b58c55f0161707b90566b72/Makefile#L78)

Reply:
The way core kubernetes components are versioned in go.mod is via minor version updates, that translate to the major versions. So the v0.23.1 tag of the compoenent is actually representing v1.23.1 of kubernetes which is fairly recent (1.24 is currently the head).

Reply:
Huh, is there a straightforward way to do the translation?  After googling for a while I can't find one. I could just `sed` out the leading 0 and replace it with a 1 but that seems like a bit of a hack since presumably at some point K8 will move on to major version 2.

Reply:
That makefile shell command specifically looks for a 0 and replaces it with a 1 (it also replaces that patch version with `x`, which is essentially a wildcard as far as the setup-envtest tool is concerned. However it seems like that command only works in the context of a Makefile. I tried copy/pasting into my terminal and it didn't do any substitution.

TL;DR: I think its safe to just sub the major 0 version for a 1 with `s/^v0/v1/`

Reply:
Alright, fixed this now that I finished the upstream stuff.